immer.js

```diff
@@ -113,10 +113,27 @@ function immer(baseState, thunk) {
         return copy
     }
 
+  function prettify(msg){
+    const isObject = typeof msg === 'object' && !Array.isArray(msg) &&
+                     'an object';
+
+    const isEmptyString = (/ /.test(msg) || (typeof msg === 'string' && msg.length === 0)) &&
+                          'an empty string';
+
+    const prettyObjectOrEmptyStringMsg = (isObject || isEmptyString);
+
+    return prettyObjectOrEmptyStringMsg || msg;
+  }
+
     // create proxy for root
     const rootClone = createProxy(baseState)
     // execute the thunk
-    thunk(rootClone)
+    const maybeVoidReturn = thunk(rootClone)
+
+    //values either than undefined will trigger warning;
+     !Object.is(maybeVoidReturn, undefined) &&
+     console.warn(`Immer callback expects no return value. However %s was returned`, prettify(maybeVoidReturn))
```

Comment:
Since this is a pretty clear warning, I would keep it simple and just use `typeof maybeVoidReturn` without prettiefying.

Reply:
I did that initially . But null ,objects , arrays all printed object . I was thinking I could be more specific. I would use type of instead then